src/navigator.js

```diff
@@ -211,6 +219,19 @@ Navigator.prototype = {
     this.navigate(uri);
   },
 
+  getExitMessage: function () {
+    return compact(map(this._emitters, function (emitter) {
+      return emitter.getExitMessage();
+    })).join(', ') || undefined;
+  },
```

Comment:
I'm slightly concerned about the `join(', ')` call resulting in a string that could be less operable than an array of values. Although I may be unclear on the underlying assumptions of the system?